homeassistant/components/camera/onvif.py

```diff
@@ -167,6 +183,12 @@ def async_added_to_hass(self):
     def async_camera_image(self):
         """Return a still image response from the camera."""
         from haffmpeg import ImageFrame, IMAGE_JPEG
+
+        if not self._input:
+            self.obtain_input_uri()
```

Comment:
You are not allowed to call methods that do I/O from an async context. You have to run this in the worker pool:

```python
await self.hass.async_add_job(self.obtain_input_uri)
```